content/blogs/newsletter-102.md

```diff
@@ -0,0 +1,131 @@
+and access tokens.
+
+An access check is an action done by the kernel to verify that if an object can be accessed by a requestor. If the requestor does not have the right privileges to access
+an object or the object itself prevents others from doing certain actions imposed by an access control list (ACL), access is denied. There was a hack in `SepAccessCheck` where access
+was granted to anybody, regardless of the security restrictions imposed by security descriptors. This could lead to bad actors tamper and toy around with objects.
+
+Thanks to his work, the hack in `SepAccessCheck` has been removed and security access checks are now operational and the kernel can prevent any unauthorized access from whoever tries
+to access an object with improper access rights. His future plan for the Security Subsystem is to implement security auditing of all objects and miscellaneous improvements where needed.
+
+## Registry improvements
+For ReactOS to reach Beta phase it has to solve two major milestones -- stability and corruptions. Usually corruptions in ReactOS can manifest in form of filesystem and registry corruptions.
+While filesystem corruptions can be prevented by choosing a more modern filesystem such as BtrFS which is what we currently support, registry corruptions are by themselves a death
+sentence as ReactOS currently doesn't implement any self-healing mechanisms to recover the system from corrupted registry hives. As a result, the user is forced to do a clean reinstallation of the system.
+In worst cases, the system would still boot up but stuck in the login prompt dialog (aka the "Ctrl+Alt+Del" dialog) as the user profile has been corrupted or missing. The latter generally happens when one force shutdowns ReactOS after first clean install.
+
+George Bișoc is currently working on improving the registry by implementing the necessary registry self-healing mechanisms and `CmCheckRegistry`, a fundamental function that validates the registry
+for any corrupted parts of the registry and takes appropriate actions at recovering a registry hive from such corrupt parts. While his work is currently in progress ([#4571](https://github.com/reactos/reactos/pull/4571)), so far he has received some positive results from testers. A small piece of his work has been merged, which fixes the registry flusher of ReactOS not working for almost a decade! This fix also
+solves the problem of the missing user profile due to force shutdown of the system, as such you should no longer expect the infamous "Ctrl+Alt+Del" dialog prompt.
+
+In addition he is working on the registry cache implementation ([#5088](https://github.com/reactos/reactos/pull/5088)), as ReactOS doesn't properly sync the cached information thus leading to inconsistent results.
+
+Currently most of this work is still upheld because it would need some code refactoring in our bootloader, FreeLdr, as the implemented core registry fixup routines bloats the bootloader.
```

Comment:
Bloat is something technically unneeded (like eye-candy stuff), so in this case it's not bloat, it just makes its size more than wanted:
```suggestion
Currently most of this work is still upheld because it would need some code refactoring in our bootloader, FreeLdr, as the implemented core registry healing routines makes bootloader size larger than we want it to be.
```

Reply:
'larger than we want it to be' --> It can't boot with a too big bootloader, so it is not really wanting, but a necessity to bring size down.

Reply:
Yes, this would be more precise. :+1: